### PR TITLE
 check inputstream when deserialize object

### PR DESCRIPTION
**Pull request by glorysdj**
please refer to 
https://www.owasp.org/index.php/Deserialization_of_untrusted_data
https://www.contrastsecurity.com/security-influencers/java-serialization-vulnerability-threatens-millions-of-applications

---
This change is [Reviewable](https://reviewable.io/reviews/intel-analytics/analytics-zoo/1200)

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceModel.scala [hunk 1]
```diff
@@ -40,42 +38,46 @@ class InferenceModel(private var supportedConcurrentNum: Int = 1,
...
     this.originalModel = InferenceModelFactory.loadFloatModel(modelPath, weightPath)
     offerModelQueue()
   }
 
   /**
-   * loads a caffe model
-   * @param modelPath the path of the prototxt file
-   * @param weightPath the path of the caffemodel file
-   */
+    * loads a caffe model
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceModel.scala]: ??? style error

Comment (glorysdj) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceModel.scala]: ye... workjing on it

**Comment by hkvision** [comment]
CheckedObjectInputStream.scala Looks great to me :)

**Comment by glorysdj** [comment]
http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/1988/
jenkins passed

**Pull request merged**